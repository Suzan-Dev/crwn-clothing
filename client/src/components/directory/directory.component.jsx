import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import './directory.style.scss';

import MenuItem from '../menu-item/menu-item.component';
import { selectDirectorySections } from '../../redux/directory/directory.selector';

export const Directory = ({ sections }) => (
  <div className='directory-menu'>
    {sections.map(({ id, ...otherSectionsProps }) => (
      <MenuItem key={id} {...otherSectionsProps} />
    ))}
  </div>
);

const mapStateToProps = createStructuredSelector({
  sections: selectDirectorySections,
});

export default connect(mapStateToProps)(Directory);
