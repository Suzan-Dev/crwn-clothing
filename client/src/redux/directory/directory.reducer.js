export const INITIAL_STATE = {
  sections: [
    {
      title: 'hats',
      imageUrl:
        'https://res.cloudinary.com/cloudimagex/image/upload/v1623552823/Crwn-Clothing/Homepage/hats_n8dibu.jpg',
      id: 1,
      linkUrl: 'shop/hats',
    },
    {
      title: 'jackets',
      imageUrl:
        'https://res.cloudinary.com/cloudimagex/image/upload/v1623552831/Crwn-Clothing/Homepage/jackets_xi53xp.jpg',
      id: 2,
      linkUrl: 'shop/jackets',
    },
    {
      title: 'sneakers',
      imageUrl:
        'https://res.cloudinary.com/cloudimagex/image/upload/v1623553049/Crwn-Clothing/Homepage/sneakers_azozgi.jpg',
      id: 3,
      linkUrl: 'shop/sneakers',
    },
    {
      title: 'womens',
      imageUrl:
        'https://res.cloudinary.com/cloudimagex/image/upload/v1623552841/Crwn-Clothing/Homepage/womens_u1x8ee.jpg',
      size: 'large',
      id: 4,
      linkUrl: 'shop/womens',
    },
    {
      title: 'mens',
      imageUrl:
        'https://res.cloudinary.com/cloudimagex/image/upload/v1623553401/Crwn-Clothing/Homepage/mens_okoivq.jpg',
      size: 'large',
      id: 5,
      linkUrl: 'shop/mens',
    },
  ],
};

const directoryReducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    default:
      return state;
  }
};

export default directoryReducer;
