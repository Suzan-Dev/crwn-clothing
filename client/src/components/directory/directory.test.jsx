import React from 'react';
import { shallow } from '../../setupTests';
import { Directory } from './directory.component';

describe('Testing: Directory component!', () => {
  const mockSections = [{ id: 1 }, { id: 2 }];

  const mockProps = {
    sections: mockSections,
  };

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Directory {...mockProps} />);
  });

  test('should render Directory component.', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should render num of withRouter Directory component as in sections array.', () => {
    expect(wrapper.find('MenuItem').length).toBe(mockSections.length);
  });
});
