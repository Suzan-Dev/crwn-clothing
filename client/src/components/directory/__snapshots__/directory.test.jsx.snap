// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing: Directory component! should render Directory component. 1`] = `
"<div className=\\"directory-menu\\">
  <MenuItem />
  <MenuItem />
</div>"
`;
